sklearn/model_selection/split.py

```diff
@@ -0,0 +1,1106 @@
+    >>> X = np.array([[1, 2], [3, 4], [5, 6], [7, 8]])
+    >>> y = np.array([1, 2, 1, 2])
+    >>> labels = np.array([1, 1, 2, 2])
+    >>> lol = LeaveOneLabelOut()
+    >>> print(lol)
+    sklearn.model_selection.split.LeaveOneLabelOut()
+    >>> for train_index, test_index in lol.split(X, y, labels):
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    ...    print(X_train, X_test, y_train, y_test)
+    TRAIN: [2 3] TEST: [0 1]
+    [[5 6]
+     [7 8]] [[1 2]
+     [3 4]] [1 2] [1 2]
+    TRAIN: [0 1] TEST: [2 3]
+    [[1 2]
+     [3 4]] [[5 6]
+     [7 8]] [1 2] [1 2]
+
+    """
+
+    def _pre_split_check(self, X, y=None, labels=None):
+        return
+
+    def _iter_test_masks(self, X, y, labels):
+        # We make a copy of labels to avoid side-effects during iteration
+        labels = np.array(labels, copy=True)
+        unique_labels = np.unique(labels)
+        for i in unique_labels:
+            yield labels == i
+
+    def __repr__(self):
+        return '%s.%s()' % (
+            self.__class__.__module__,
+            self.__class__.__name__,
+        )
+
+
+class LeavePLabelOut(_PartitionIterator):
+    """Leave-P-Label_Out cross-validation iterator
+
+    Provides train/test indices to split data according to a third-party
+    provided label. This label information can be used to encode arbitrary
+    domain specific stratifications of the samples as integers.
+
+    For instance the labels could be the year of collection of the samples
+    and thus allow for cross-validation against time-based splits.
+
+    The difference between LeavePLabelOut and LeaveOneLabelOut is that
+    the former builds the test sets with all the samples assigned to
+    ``p`` different values of the labels while the latter uses samples
+    all assigned the same labels.
+
+    Read more in the :ref:`User Guide <split>`.
+
+    Parameters
+    ----------
+    p : int
+        Number of samples to leave out in the test split.
+
+    Examples
+    --------
+    >>> from sklearn.model_selection import LeavePLabelOut
+    >>> X = np.array([[1, 2], [3, 4], [5, 6]])
+    >>> y = np.array([1, 2, 1])
+    >>> labels = np.array([1, 2, 3])
+    >>> lpl = LeavePLabelOut(p=2)
+    >>> print(lpl)
+    sklearn.model_selection.split.LeavePLabelOut(p=2)
+    >>> for train_index, test_index in lpl.split(X, y, labels):
+    ...    print("TRAIN:", train_index, "TEST:", test_index)
+    ...    X_train, X_test = X[train_index], X[test_index]
+    ...    y_train, y_test = y[train_index], y[test_index]
+    ...    print(X_train, X_test, y_train, y_test)
+    TRAIN: [2] TEST: [0 1]
+    [[5 6]] [[1 2]
+     [3 4]] [1] [1 2]
+    TRAIN: [1] TEST: [0 2]
+    [[3 4]] [[1 2]
+     [5 6]] [2] [1 1]
+    TRAIN: [0] TEST: [1 2]
+    [[1 2]] [[3 4]
+     [5 6]] [1] [2 1]
+    """
+
+    def __init__(self, p):
```

Comment:
In this case I think we can safely do the renaming without ambiguity. Assuming the docstring is clear.